dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/web/SelectionPageGenerator.java

```diff
@@ -276,9 +279,38 @@ String createJavascript() {
     }
 
     String getHtmlTitle() {
-        return Tr.formatMessage(tc, request.getLocales(), "SELECTION_PAGE_TITLE");
+        return Tr.formatMessage(tc, getLocale(request.getLocales()), "SELECTION_PAGE_TITLE");
     }
 
+    /**
+     * Determines the preferred Locale of the request, <b>as supported by the Liberty profile</b>.
+     * In other words, if the most-preferred Locale that is requested that is not supported by the
+     * Liberty runtime, then the next most-preferred Locale will be used, finally resulting in the
+     * JVM's default Locale.
+     * <p>
+     * The net effect of this is any French locale (fr, fr_ca, fr_fr, etc) would resolve to just 'fr'.
+     * Any Portugese locale ('pt') other than Brazillian ('pt_br') would resolve to the JVM default
+     * encoding. Portugese Brazillian is tranlated to, so 'pt_br' is returned. Any English locale
+     * is returned as 'en'. Any unrecognized locale resolves to the JVM default.
+     * 
+     * @return The Locale for the request. The best match supported by the Liberty runtime is returned, or the defualt Locale.
+     */
+    public static Locale getLocale(final Enumeration<Locale> locales) {
```

Comment:
So this would make three duplicates of this `getLocale()` method. Instead of duplicating the method so many times, adding it to a single common location - for instance a new `lang` package under `com.ibm.ws.security.common` - would be ideal. Even adding it to a single common location in security would still be duplicating code since we're just copying this method from a UI project. The theoretical ideal is a single method that the entire product can use, but off the top of my head I don't know where I'd put something like that.